packages/menu/src/use-menu.ts

```diff
@@ -431,7 +434,7 @@ export function useMenuItem(props: UseMenuItemProps) {
         focus(ref.current)
       }
     } else {
-      if (document.activeElement !== menuRef.current) {
+      if (document.activeElement !== menuRef.current && isOpen) {
```

Comment:
since the menu is no longer being unmounted from the page we have to check if `isOpen` to avoid focusing on a not visible `MenuList`